gluish/common.py

```diff
@@ -64,3 +113,202 @@ def run(self):
+                result = False
+
+        return result
+
+    def output(self):
+        return luigi.LocalTarget(path=os.path.join(str(self.basedirectory), str(self.repositorydirectory)))
+
+
+class CleanSolrIndex(luigi.Task):
+    """
+    Cleans a given Solr index from already existing data of a data package
+    """
+    date = ClosestDateParameter(default=datetime.date.today())
+    solruri = luigi.Parameter()
+    solrcore = luigi.Parameter()
+    deletequery = luigi.Parameter()
+    taskdir = luigi.Parameter()
+    salt = luigi.Parameter()
+    uritemplate = '{solruri}{solrcore}/update?commit=true'
+    headers = {'Content-Type': 'text/xml'}
+    payloadtemplate = '<delete><query>{deletequery}</query></delete>'
+
+    def requires(self):
+        return Available(service=self.solruri,
+                         message="provide a running Solr, please")
+
+    def run(self):
+        uri = self.uritemplate.format(solruri=self.solruri,
+                                      solrcore=self.solrcore)
+        payload = self.payloadtemplate.format(deletequery=self.deletequery)
+        result = {}
+
+        try:
+            response = requests.post(uri, data=payload, headers=self.headers)
+
+            result['status_code'] = response.status_code
+            result['content'] = response.content.decode('utf-8')
+
+            if result['status_code'] != 200:
+                raise RuntimeError("""could not successfully clean Solr core '%s' (status code = '%d')""" % (
+                    uri, response.status_code))
+
+            if not str(result['content']).startswith(
+                    """<?xml version="1.0" encoding="UTF-8"?>\n<response>\n<lst name="responseHeader"><int name="status">0</int>"""):
+                raise RuntimeError(
+                    """could not successfully clean Solr core '%s' (response = '%s')""" % (uri, result['content']))
+        except requests.exceptions.ConnectionError as error:
+            result['status_code'] = 404
+            result['error'] = str(error).replace('"', '\\"')
+
+            raise RuntimeError("""could not successfully establish connection to Solr index '%s'""" % uri)
+        finally:
+            with self.output().open('w') as outfile:
+                outfile.write(json.dumps(result, indent=None) + "\n")
+
+    def output(self):
+        filename = 'response_' + str(self.salt) + '.json'
+        return luigi.LocalTarget(path=os.path.join(self.taskdir, filename))
+
+    def complete(self):
+        # TODO maybe define a better complete criteria (?) (background: complete() will be executed before run())
+        # e.g. check, whether something of this source is in the index
+        # or move all the stuff from run to complete (?)
+        if not self.output().exists():
+            return False
+
+        with self.output().open('r') as resultfile:
+            result = json.loads(resultfile.read())
+
+            if result['status_code'] != 200:
```

Comment:
hm, funny interpretation of "local" in Python :)

Reply:
Yes, it can be confusing - but only for a moment :) -- many people (including me) wondered about this, though:

* https://stackoverflow.com/a/292502/89391